File: src/main/java/sprykerFileCreator/model/generator/DocBlockGenerator.java
package pav.sprykerFileCreator.model.generator;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.definition.DefinitionProviderInterface;
import pav.sprykerFileCreator.model.renderer.dto.DocBlockItem;
import pav.sprykerFileCreator.model.renderer.dto.PhpClassInterface;
import pav.sprykerFileCreator.model.resolver.ClassResolverInterface;

import java.util.ArrayList;
import java.util.List;

public class DocBlockGenerator implements DocBlockGeneratorInterface {

    private ClassResolverInterface classResolver;
    private DefinitionProviderInterface definitionProvider;

    public DocBlockGenerator(
            ClassResolverInterface classResolver,
            DefinitionProviderInterface definitionProvider
    ) {
        this.classResolver = classResolver;
        this.definitionProvider = definitionProvider;
    }

    public List<DocBlockItem> getDocBlockItems(
            String[] classTypes,
            String bundleName
    ) throws Exception {
        List<DocBlockItem> docBlockItemList = new ArrayList<>();

        for (String classType : classTypes) {
            PhpClassInterface returnClass = this.classResolver.resolveBundleClass(classType, bundleName);

            if (returnClass != null) {
                String returnClassName = returnClass.getFullQualifiedName();
                String docBlockMethod = this.getMethodForType(classType);

                DocBlockItem docBlockItem = new DocBlockItem(
                        "method",
                        returnClassName,
                        docBlockMethod
                );

                docBlockItemList.add(docBlockItem);
            }
        }

        return docBlockItemList;
    }

    private String getMethodForType(String classType) throws Exception {
        ClassDefinitionInterface classDefinition = this.definitionProvider.getDefinitionByType(classType);

        return classDefinition.getMethodForReturnType();
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/generator/DocBlockGenerator.kt
package com.github.tobi812.sprykerplugin.models.generator

import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface
import com.github.tobi812.sprykerplugin.models.definitions.DefinitionProviderInterface
import com.github.tobi812.sprykerplugin.models.renderer.dto.DocBlockItem
import com.github.tobi812.sprykerplugin.models.renderer.dto.PhpClassInterface
import com.github.tobi812.sprykerplugin.models.resolver.ClassResolverInterface
import com.intellij.openapi.components.service
import com.intellij.openapi.project.Project
import java.lang.Exception
import java.util.ArrayList

class DocBlockGenerator(
    private val project: Project
) : DocBlockGeneratorInterface {

    @Throws(Exception::class)
    override fun getDocBlockItems(
        classTypes: Array<String>,
        bundleName: String
    ): List<DocBlockItem> {
        val classResolver = project.service<ClassResolverInterface>()
        val docBlockItemList: MutableList<DocBlockItem> = ArrayList<DocBlockItem>()
        for (classType in classTypes) {
            val returnClass: PhpClassInterface? = classResolver.resolveBundleClass(classType, bundleName)
            if (returnClass != null) {
                val returnClassName: String = returnClass.getFullQualifiedName()
                val docBlockMethod = this.getMethodForType(classType)
                val docBlockItem = DocBlockItem(
                    "method",
                    returnClassName,
                    docBlockMethod
                )
                docBlockItemList.add(docBlockItem)
            }
        }

        return docBlockItemList
    }

    @Throws(Exception::class)
    private fun getMethodForType(classType: String): String {
        val classDefinition: ClassDefinitionInterface = project.service<DefinitionProviderInterface>()
            .getDefinitionByType(classType)

        return classDefinition.methodForReturnType
    }
}
